cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1043,21 +1026,46 @@ private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> onReady(Reconcilia
                                 new InvalidResourceException(CruiseControlIssues.cruiseControlDisabled.getMessage())));
                     }
 
+                    Future<Void> resourcePatchFuture = Future.future(future -> {
+                        if (kafkaRebalance.getStatus() != null
+                                && kafkaRebalance.getStatus().getObservedGeneration() != kafkaRebalance.getMetadata().getGeneration()) {
+
+                            KafkaRebalanceBuilder patchedKafkaRebalance = new KafkaRebalanceBuilder(kafkaRebalance);
+
+                            patchedKafkaRebalance
+                                    .editMetadata()
+                                        .addToAnnotations(Map.of(ANNO_STRIMZI_IO_REBALANCE, KafkaRebalanceAnnotation.refresh.toString()))
+                                    .endMetadata();
+
+                            kafkaRebalanceOperator.patchAsync(reconciliation, patchedKafkaRebalance.build()).onComplete(
+                                    r -> {
+                                        if (r.succeeded()) {
+                                        LOGGER.debugCr(reconciliation, "The KafkaRebalance resource is updated with refresh annotation");
+                                            future.complete();
+                                        } else {
+                                        LOGGER.errorCr(reconciliation, "Failed to update the KafkaRebalance resource with refresh annotation");
+                                        future.fail(r.cause());
+                                        }
+                                    });
+                        } else {
+                            future.complete();
+                        }
+                    });
```

Comment:
I think that normally, we would do it something like this:

```suggestion
                    Future<Void> resourcePatchFuture;
                    if (kafkaRebalance.getStatus() != null
                            && kafkaRebalance.getStatus().getObservedGeneration() != kafkaRebalance.getMetadata().getGeneration()) {
                        KafkaRebalanceBuilder patchedKafkaRebalance = new KafkaRebalanceBuilder(kafkaRebalance)
                                    .editMetadata()
                                        .addToAnnotations(Map.of(ANNO_STRIMZI_IO_REBALANCE, KafkaRebalanceAnnotation.refresh.toString()))
                                    .endMetadata()
                                    .build()

                        resourcePatchFuture = kafkaRebalanceOperator.patchAsync(reconciliation, patchedKafkaRebalance);
                    } else {
                        resourcePatchFuture = Future.succeededFuture();
                    }
```

(I did not test it or wrote it in an IDE, so it might be misaligned or buggy ... but it should give you an idea :-D)